src/Moryx.Runtime.Kernel/FileSystem/LocalFileSystem.cs

```diff
@@ -0,0 +1,278 @@
+﻿using Castle.MicroKernel.Registration;
+using Microsoft.Extensions.Logging;
+using Moryx.FileSystem;
+using Moryx.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Moryx.Runtime.Kernel.FileSystem
+{
+    internal class LocalFileSystem : IMoryxFileSystem
+    {
+        private string _fsDirectory;
+        private string _ownerFilesDirectory;
+        private readonly ILogger _logger;
+
+        private readonly List<MoryxFileTree> _ownerTrees = new List<MoryxFileTree>();
+
+        public LocalFileSystem(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger(nameof(LocalFileSystem));
+        }
+
+        public void SetBasePath(string basePath = "fs")
+        {
+            _fsDirectory = Path.Combine(Directory.GetCurrentDirectory(), basePath);
+            _ownerFilesDirectory = Path.Combine(_fsDirectory, "owners");
+        }
+
+        public void LoadTrees()
```

Comment:
Who calls this? Should the FileSystem maybe be an `IHostedService` to have a start lifecycle hook which loads the trees? 🤔

Reply:
Good point